pkgs/build-support/docker/default.nix

```
@@ -1072,6 +1072,8 @@ rec {
       command ? null
     , # Same as `command`, but runs the command in a non-interactive shell instead. See `--run` in `man nix-shell`
       run ? null
+    , # Additional packages to add to the image, these will also be hardcoded into the exposed PATH
+      extraPackages ? []
```

```suggestion
      extraPATH ? []
```

Packages doesn't tell me how it's used.

---

Still a bad name though. @infinisil what is the *meaning* of staticPath? How is it intended to be used, if at all? What would be a good name for this parameter?

---

I think extraPackages is fine. We may also install other things from packages in the future, not just binaries.

---

And staticPath is internal anyways, it's just what PATH is set to initially

---

Hmm thinking about this again, you're right Robert, this is a bit of an odd feature. Generally build inputs should be used to add things to PATH. Here it's only needed because of jenkins doing something weird with calling binaries.


I wonder @fionera, what's the minimal possible interface to support this use case? Could there just be a single `jenkinsWorkaround = true` that adds `nohup` to the PATH for example?

---

In my case I dont need it for Jenkins but for Kubernetes where I need my buildFHSUserEnv output in the static PATH variable

---

Hmm so kind of the main reason for `buildNixShellImagine` is for it to source the stdenv setup stuff and make it look very similar to a `nix-shell`, therefore also ensuring the binaries from `buildInputs` are on `PATH`, etc. But it seems like if you provide a custom command, that might not source the appropriate files to imitate `nix-shell` in the same way. Or how would that still work?

---

I provide the derivation that creates the FHSEnv as drv but also in extraPackages. In its shell parameter it has a script that executes everything which is given as parameter as normal command. Thats my workaround for using FHSEnv for arbitrary commands directly.

```
...
pkgs.dockerTools.buildNixShellImage {
  inherit drv;
  extraPackages = [
    drv
  ];
...
```

`monogon-nix -- /monogon-infra/k8s/apps/jenkins/build/agentlauncher`

---

Do you really depend on the `nix-shell` like behavior then? Because if not, you could use a different docker builder like [`buildLayeredImage`](https://nixos.org/manual/nixpkgs/unstable/#ssec-pkgs-dockerTools-buildLayeredImage) instead, that allows you to set everything, including env vars

---

Yeah works for me